### PR TITLE
fix Rectifier model parsing, closes lp-research/lpslam_node#18

### PR DESCRIPTION
**Pull request by ItsJas**
This will enable stereo rectification OpenVSLAM side and fix [this issue](https://github.com/lp-research/lpslam_node/issues/18).

### CODE REVIEW

File: src/openvslam/system.cc [hunk 1]
```diff
@@ -115,11 +116,17 @@ system::system(const std::shared_ptr<config>& cfg, const std::string& vocab_file
...
 
-    // create retifier, make optional
-    if (cfg_->yaml_node_["StereoRectifier.model"]) {
-        rectifier_ = std::make_unique<util::stereo_rectifier>(cfg_);
+    // create rectifier, make optional
+    if (cfg_->yaml_node_["StereoRectifier"]) {
+    if (cfg_->yaml_node_["StereoRectifier"]["model"]) {
+        spdlog::info("rectifier model found, going to rectify");
+        rectifier_ = std::make_unique<util::stereo_rectifier>(
+        cfg->camera_, openvslam::util::yaml_optional_ref(cfg->yaml_node_, "StereoRectifier"));
```
Comment (AlexeyMerzlyakov) [inline src/openvslam/system.cc]: Please fix the formatting to be something like:
```
        rectifier_ = std::make_unique<util::stereo_rectifier>(
            cfg->camera_, openvslam::util::yaml_optional_ref(cfg->yaml_node_, "StereoRectifier"));
```

Comment (AlexeyMerzlyakov) [inline src/openvslam/system.cc]: Why `util::stereo_rectifier()` constructor was changed from one arguments constructor to 2-args one? If this fixes the problem that `stereo_rectifier(cfg->camera_, cfg->yaml_node_)` is not being called correctly, why don't fix it in the `stereo_rectifier()` constructor itself? BTW, it looks like it already [was made](https://github.com/stella-cv/stella_vslam/blob/main/src/stella_vslam/util/stereo_rectifier.cc#L14) in Community OpenVSLAM.

File: src/openvslam/system.cc [hunk 2]
```diff
@@ -115,11 +116,17 @@ system::system(const std::shared_ptr<config>& cfg, const std::string& vocab_file
...
+    }
+    }
+    else{
+        spdlog::info("failed to load rectifier, skipping this step");
     }
-
```
Comment (AlexeyMerzlyakov) [inline src/openvslam/system.cc:122]: Unnecessary style line removal, in my opinion

File: src/openvslam/system.cc (hunk 1 above)
Comment (ItsJas) [reply src/openvslam/system.cc]: Good call! I fixed it this way with reference to the examples like [here](https://github.com/lp-research/openvslam/blob/237dcaae25785353c3c97de3cfd835a278e86da8/example/run_euroc_slam.cc#L163) thinking it's the right way to call it but it's better to do it [stereo_rectifier side](https://github.com/lp-research/openvslam/blob/237dcaae25785353c3c97de3cfd835a278e86da8/src/openvslam/util/stereo_rectifier.cc#L12) as you suggested!

File: src/openvslam/system.cc [hunk 3]
```diff
@@ -115,9 +115,14 @@ system::system(const std::shared_ptr<config>& cfg, const std::string& vocab_file
...
-    if (cfg_->yaml_node_["StereoRectifier.model"]) {
-        rectifier_ = std::make_unique<util::stereo_rectifier>(cfg_);
+    // create rectifier, make optional
+    if (cfg_->yaml_node_["StereoRectifier"]) {
+        if (cfg_->yaml_node_["StereoRectifier"]["model"]) {
+            spdlog::info("rectifier model found, going to rectify");
+            rectifier_ = std::make_unique<util::stereo_rectifier>(cfg_);
+        }
+    } else {
+        spdlog::info("failed to load rectifier, skipping this step");
```
Comment (AlexeyMerzlyakov) [inline src/openvslam/system.cc]: In my opinion, the message is better to be something like `rectifier model not found in the config, skipping this step`. And this also could be a `spdlog::warn()`. No rushing this.

**Review by AlexeyMerzlyakov — APPROVED** [review]
One nitpick from my side. Overall, the commit is good for merge.

File: src/openvslam/system.cc (hunk 3 above)
Comment (ItsJas) [reply src/openvslam/system.cc]: The warning message you suggested describes the issue better. I will change it!

File: src/openvslam/system.cc [hunk 4]
```diff
@@ -115,9 +115,14 @@ system::system(const std::shared_ptr<config>& cfg, const std::string& vocab_file
     global_optimizer_ = new global_optimization_module(map_db_, bow_db_, bow_vocab_, cfg_->yaml_node_, camera_->setup_type_ != camera::setup_type_t::Monocular);
 
 
-    // create retifier, make optional
-    if (cfg_->yaml_node_["StereoRectifier.model"]) {
```
Comment (ysujith728) [inline src/openvslam/system.cc:119]: Nice improvement in checking for the parent StereoRectifier node before accessing model. This avoids potential YAML access errors and makes the configuration parsing much safer.

**Pull request merged**